packages/services/src/view-service/fees.ts

```diff
@@ -60,9 +85,46 @@ export const extractAltFee = async (
   if (auctionWithdrawAsset) {
     const withdrawAuction = await indexedDb.getAuction(auctionWithdrawAsset);
     if (withdrawAuction.auction?.input?.assetId) {
-      return withdrawAuction.auction.input.assetId;
+      const checkAssetBalance = await indexedDb.getAssetTokenMetadata(
+        request.source,
+        withdrawAuction.auction.input.assetId,
+      );
+      if (checkAssetBalance) {
+        return withdrawAuction.auction.input.assetId;
+      } else {
+        const assetId = await retrieveAltAsset(request, indexedDb);
+        if (assetId) {
+          return assetId;
+        }
+      }
+    }
+  }
```

Comment:
This looks like the equivalent to above. Maybe abstract to a helper function instead?

Reply:
removed `stakingTokenAssetId` and made `hasTokenBalance` a helper function.